compiler/packages/babel-plugin-react-compiler/src/HIR/CollectHoistablePropertyLoads.ts

```diff
@@ -111,14 +105,43 @@ export function collectHoistablePropertyLoads(
       }
     }
   }
-  const nodes = collectNonNullsInBlocks(fn, {
-    temporaries: actuallyEvaluatedTemporaries,
+  return collectHoistablePropertyLoadsImpl(fn, {
+    temporaries,
     knownImmutableIdentifiers,
```

Comment:
We can now directly pass down temporaries as we don't need to filter out artificial `LoadLocal` and `PropertyLoads` created for the now-deleted `LoweredFunction.dependencies`